ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -15840,25 +15840,33 @@ private QueryResultsCache.QueryInfo createCacheQueryInfoForQuery(QueryResultsCac
   /**
    * Some initial checks for a query to see if we can look this query up in the results cache.
    */
-  private boolean queryTypeCanUseCache() {
-    if (this.qb == null || this.qb.getParseInfo() == null) {
+  private boolean queryTypeCanUseCache(QB qbParam) {
```

Comment:
minor: I would have used simply `qb`. I know it hides the instance param, but maybe we could refactor similar stuff in diff PR